### PR TITLE
How to embed a Compose for Desktop application in a Swing-based application

### LINKED ISSUE #6712: How to Embed Desktop Compose into a Swing-based application
#### Proposed title of article
How to Embed Desktop Compose into a Swing-based application
### Writing Sample(s)
https://www.section.io/engineering-education/optimizing-and-securing-android-applications-with-r8/

#### Proposed article introduction
Compose Desktop[Kotlin] simplifies and accelerates the development of UI that may be shared between Android, web, and desktop in the near future.
Compose has the capability to interoperate with designing user interfaces with Java/Kotlin AWT and Swing.
As a result, migrating your current Swing Desktop Application to Compose Desktop may be simpler. 

This tutorial will teach a learner on how to use ComposePanel and SwingPanel in a desktop application. 

#### Key takeaways 
- A quick recap of building Swing and Desktop Compose Applications
- Using ComposePanel to create UIs
- Adding a Swing component to CFD composition

#### Article quality
This article will be very helpful to those developers who are not yet into Compose Multiplatform and would like to use some features of it building Desktop/Android applications. This will also be of help to anyone who may want to migrate his Swing application to Compose Desktop in near future.

#### References
https://www.jetbrains.com/lp/compose-desktop/

### PR DESCRIPTION
**Pull request by kariuki-boniface**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6712

### CODE REVIEW

File: content/articles/how-to-embed-compose-desktop-application-to-swing-based-application/index.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+                    ){
+                        Button(onClick = {}, shape = RoundedCornerShape(10.dp)){
+                            Text("Button 1")
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
```
Comment (john-amiscaray) [inline content/articles/how-to-embed-compose-desktop-application-to-swing-based-application/index.md:176]: Once again, please include the import statements to make it clear which classes you are importing and from where

Comment (john-amiscaray) [inline content/articles/how-to-embed-compose-desktop-application-to-swing-based-application/index.md:176]: I would recommend adding comments to this function so beginners can more easily understand it.

File: content/articles/how-to-embed-compose-desktop-application-to-swing-based-application/index.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+                factory = {
+                    val button = JButton("Swing Button")
+                    button.preferredSize = Dimension(40,40)
+                    val panel = JPanel()
+                    panel.add(button, BoxLayout.X_AXIS)
+                }
+            )
+        }
+    }
+}
```
Comment (john-amiscaray) [inline content/articles/how-to-embed-compose-desktop-application-to-swing-based-application/index.md:219]: I would also consider adding comments here for beginners to better understand it

**Review by john-amiscaray — CHANGES_REQUESTED** [review]
Here are some changes to implement before I wrap up the review. I don't believe these should take too long so I'll keep the pull request open. Just please be diligent to make these changes as not to take up room in the queue.

**Comment by kariuki-boniface** [comment]
@john-amiscaray Thank you for the heads up!. Lemme work on the changes

**Comment by LinusMuema** [comment]
plagiarism check done ✅ 

Only imports and composable bring up the red flags.

**Comment by john-amiscaray** [comment]
@kariuki-boniface Looks good. I just made some minor changes right now. Be sure to go through them in case I did something wrong or to learn from my corrections.

**Review by john-amiscaray — APPROVED** [review]

**Comment by john-amiscaray** [comment]
@kariuki-boniface Great work. We are ready for the final review now.

**Comment by hectorkambow** [comment]
@section-engineering-education/content-moderator-team 

Is anyone able to give this PR a final review? Its been ready for 19 days?

**Comment by hectorkambow** [comment]
I'll be jumping on this one

**Comment by hectorkambow** [comment]
@kariuki-boniface 

Please take note at all the changes made by your peer reviewer and myself.

**Comment by hectorkambow** [comment]
This article was quite difficult to review (had many grammatical issues).

**Pull request merged**